Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -421,6 +421,38 @@ namespace AzToolsFramework
             return newTemplateId;
         }
 
+        void PrefabSystemComponent::ChangeTemplateFilePath(const TemplateId& templateId, const AZ::IO::Path& filePath)
+        {
+            auto findTemplateResult = FindTemplate(templateId);
+            if (!findTemplateResult.has_value())
+            {
+                AZ_Warning(
+                    "Prefab", false,
+                    "PrefabSystemComponent::ChangeTemplateFilePath - "
+                    "Template associated by given Id '%llu' doesn't exist in PrefabSystemComponent.",
+                    templateId);
+                return;
+            }
+            Template& templateToChange = findTemplateResult->get();
+            if (templateToChange.GetFilePath() == filePath)
+            {
+                return;
+            }
+            
+            m_templateFilePathToIdMap.erase(templateToChange.GetFilePath());
+            m_templateFilePathToIdMap.emplace(AZStd::make_pair(filePath, templateId));
```

Comment:
Might be a good idea to check IsRelativePath() on the provided path here before injecting it into the system.

Reply:
This doesn't check if the file path has already been added. It's also less efficient to make the pair here. I recommend the following instead:
```
if (!m_templateFilePathToIdMap.try_emplace(filePath, templateId).second)
{
    AZ_Error(...);
    return;
}
```

Reply:
The `try_emplace` call will have added this value or returned `false` so this line is no longer needed.